### PR TITLE
feat: migrate from @salesforce/core v6 -> v7

### PR DESCRIPTION
**Pull request by daphne-sfdc**
### What does this PR do?

Migrates @salesforce/core from v6.7.4 -> v7.3.1

This upgrade means we are able to support Core v7 in our other dependencies, so the following dependencies are also migrated from Core v6 -> v7 in this PR:

1. @salesforce/source-deploy-retrieve - v10.3.3 -> v11.1.2
2. @salesforce/source-tracking - v5.1.11 -> v6.0.4
3. @salesforce/apex-node - v4.0.5 -> v6.0.0
4. @salesforce/templates - v60.1.0 -> v60.1.2
5. jsforce v2.0.0-beta.29 -> @jsforce/jsforce-node v3.2.0

Additional changes:
1. Changed the import of `@salesforce/core/testSetup` to use top-level imports
2. Replaced all usages of Interfaces with Types to match the new standards
3. Fixed the mocking of @salesforce/core in `workspaceContextUtils.test.ts` unit test to make it pass

### What issues does this PR fix or reference?
@W-15472197@

### Functionality Before
Extensions monorepo uses v6 of @salesforce/core

### Functionality After
Extensions monrepo uses v7 of @salesforce/core, which means we are now able to pull in new features from our shared libraries

### CODE REVIEW

File: packages/salesforcedx-vscode-soql/src/telemetry/index.ts [hunk 1]
```diff
@@ -23,10 +23,10 @@ export const stopTelemetry = (): Promise<void> => {
   return Promise.resolve();
 };
 
-export interface TelemetryModelJson extends JsonMap {
+export type TelemetryModelJson = JsonMap & {
```
Comment (peternhale) [inline packages/salesforcedx-vscode-soql/src/telemetry/index.ts:26]: @daphne-sfdc I don't believe JsonMap is not necessary.

**Review by peternhale — APPROVED** [review]

File: packages/salesforcedx-vscode-core/src/orgBrowser/metadataCmp.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@
  * Licensed under the BSD 3-Clause license.
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
+import { ListMetadataQuery } from '@jsforce/jsforce-node/lib/api/metadata';
```
[the remark this answers is not in the document]
Comment (daphne-sfdc) [reply packages/salesforcedx-vscode-core/src/orgBrowser/metadataCmp.ts:7]: Screenshot 2024-04-29 at 2 32 53 PM

It seems like ListMetadataQuery does exist at the top level but it refers to something else?
Screenshot 2024-04-29 at 2 33 54 PM

File: packages/salesforcedx-vscode-soql/src/telemetry/index.ts (hunk 1 above)
Comment (daphne-sfdc) [reply packages/salesforcedx-vscode-soql/src/telemetry/index.ts:26]: @peternhale The double negative means you think it is necessary and what's currently there is fine?

Comment (peternhale) [reply packages/salesforcedx-vscode-soql/src/telemetry/index.ts:26]: not necessary

File: packages/salesforcedx-vscode-soql/src/sf.ts [hunk 3]
```diff
@@ -5,12 +5,12 @@
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
+import { DescribeSObjectResult } from '@jsforce/jsforce-node';
```
Comment (gbockus-sf) [inline packages/salesforcedx-vscode-soql/src/sf.ts:8]: We already moved to jsforce-node!  🍠

**Pull request merged**